addons/point_of_sale/static/src/js/models.js

```diff
@@ -292,24 +282,32 @@ class PosGlobalState extends PosModel {
 
     // reload the list of partner, returns as a promise that resolves if there were
     // updated partners, and fails if not
-    load_new_partners(){
+    async load_new_partners(){
+        try {
+            const domain = this.prepare_new_partners_domain();
+            const partners = await this.env.services.orm.silent.call(
+                'pos.session',
+                'get_pos_ui_res_partner_by_params',
+                [[odoo.pos_session_id], { domain }]
+            );
+            if (!this.db.add_partners(partners)) {
+                throw 'Failed in updating partners.';
+            }
+        } catch (e) {
+            throw undefined;
+        }
         return new Promise((resolve, reject)  => {
```

Comment:
if it's possible, try to use await rather than a promise with then since we're in refactoring mode